src/components/PaginationContainer.ts

```diff
@@ -0,0 +1,177 @@
+                pagingStyle: this.props.pagingStyle
+            });
+        }
+
+        return null;
+    }
+
+    private findListView() {
+        if (this.state.findingListViewWidget) {
+            const queryNode = findDOMNode(this) as HTMLElement;
+            const targetNode = findTargetNode(queryNode);
+            let hideUnusedPaging = false;
+            let targetListView: ListView | null = null;
+            let listViewSize = 0;
+            let offset = 0;
+            let dataSource: ListView["_datasource"];
+
+            if (targetNode) {
+                this.hideLoadMoreButton(targetNode);
+                targetListView = dijitRegistry.byNode(targetNode);
+
+                if (targetListView) {
+                    dataSource = targetListView._datasource;
+                    listViewSize = dataSource._setSize;
+                    offset = targetListView._datasource._pageSize;
+                    hideUnusedPaging = (offset >= dataSource._setSize) && this.props.hideUnusedPaging;
+
+                    dojoAspect.after(targetListView, "_onLoad", () => {
+                        if (this.state.targetListView) {
+                            this.setState({
+                                listViewSize: this.state.targetListView._datasource._setSize,
+                                offset
+                            });
+                        }
+                    });
+                }
+            }
+
+            this.validateListView({ targetNode, targetListView, hideUnusedPaging, listViewSize, offset });
+        }
+    }
+
+    private validateListView(props: ValidateProps) {
+        const message = ValidateConfigs.validate({
+            ...this.props as WrapperProps,
+            queryNode: props.targetNode,
+            targetListView: props.targetListView
+        });
+
+        this.setState({
+            findingListViewWidget: false,
+            hideUnusedPaging: props.hideUnusedPaging,
+            listViewSize: props.listViewSize,
+            message,
+            offset: props.offset,
+            targetListView: props.targetListView,
+            targetNode: props.targetNode,
+            validationPassed: message === ""
+        });
+    }
+
+    private hideLoadMoreButton(targetNode?: HTMLElement | null) {
+        if (targetNode) {
+            const buttonNode = targetNode.querySelector(".mx-listview-loadMore") as HTMLButtonElement;
+
+            if (buttonNode) {
+                buttonNode.classList.add("widget-pagination-hide-load-more");
+            }
+
+            this.listListViewHeight = targetNode.clientHeight;
+        }
+    }
+
+    private updateListView(offSet: number) {
+        const { targetListView, targetNode, validationPassed } = this.state;
+
+        if (targetListView && targetNode && validationPassed) {
+            const listNode = targetNode.querySelector("ul") as HTMLUListElement;
+
+            listNode.style.height = `${this.listListViewHeight}px`;
```

Comment:
Is this height ever removed afterwards? (I think it should be removed after _renderData)
Please add a test case for this situation where not all items have the same height, (if this is non already)

Reply:
Done,